### PR TITLE
Http update

### PR DESCRIPTION
**Pull request by Kingdorian**
I updated the http api to be more clear. There might need to be changed a couple of things after we merge this with Naomi since I saw she also made some good changes in the code in the http package.

### CODE REVIEW

File: backend/src/main/java/com/benine/backend/Main.java [hunk 1]
```diff
@@ -57,27 +48,17 @@ public static void main(String[] args) {
...
-      server.start();
       while (true) {
         Thread.sleep(100);
       }
-    } catch (IOException e) {
-      logger.log("Unable to start server", LogEvent.Type.CRITICAL);
     } catch (InterruptedException e) {
-      logger.log("Unable to start server", LogEvent.Type.CRITICAL);
+      e.printStackTrace();
+      logger.log("Exception occured", LogEvent.Type.CRITICAL);
```
Comment (jochem725) [inline backend/src/main/java/com/benine/backend/Main.java]: I don't think this log message is clear enough. `Exception occurred` does not make clear what kind of exception was thrown.

File: backend/src/main/java/com/benine/backend/http/CameraInfoHandler.java [hunk 2]
```diff
@@ -29,18 +31,15 @@ public CameraInfoHandler(CameraController controller) {
    * @throws IOException When writing the response fails.
    */
   public void handle(HttpExchange exchange) throws IOException {
-    //TODO add logging stuff
+    getLogger().log("Got an http request with uri: " + exchange.getRequestURI(), LogEvent.Type.INFO);
     JSONObject jsonObj = new JSONObject();
     ArrayList<Camera> cameras = getCameraController().getCameras();
     JSONArray camerasJSON = new JSONArray();
     for (Camera cam : cameras) {
```
Comment (jochem725) [inline backend/src/main/java/com/benine/backend/http/CameraInfoHandler.java]: Maybe the camerasJSON object can be returned by a separate method, to split handling from generating the JSON a bit more.

File: backend/src/main/java/com/benine/backend/http/HttpController.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    server.createContext("/camera/", new CameraInfoHandler(camController, logger));
+    for (Camera cam : camController.getCameras()) {
+      createHandlers(cam);
+    }
+  }
+  /**
+   * Creates the handlers for a certain camera.
+   * @param cam camera to create handlers for.
+   */
+  public void createHandlers(Camera cam) {
```
Comment (jochem725) [inline backend/src/main/java/com/benine/backend/http/HttpController.java]: Should this method be public? It is only used locally as far as I can see.

Comment (Kingdorian) [inline backend/src/main/java/com/benine/backend/http/HttpController.java]: Jep that slipped through, I'll fix it.

File: backend/src/main/java/com/benine/backend/http/CameraInfoHandler.java (hunk 2 above)
Comment (Kingdorian) [inline backend/src/main/java/com/benine/backend/http/CameraInfoHandler.java]: Great idea I'll create a getCamerasJSON method in the cameraController even.

File: backend/src/main/java/com/benine/backend/Main.java (hunk 1 above)
Comment (Kingdorian) [inline backend/src/main/java/com/benine/backend/Main.java]: Good point.

**Comment by jochem725** [comment]
Looks great, I've added some comments. Also check PMD, Findbugs and Checkstyle, as they have some solvable warnings.

**Comment by bvanwijk** [comment]
Looks good, if you have fixed Jochems comments than I agree with merging.

**Comment by NaomidR** [comment]
This indeed looks good and I agree with merging after you have fixed Jochems comments.

**Comment by Kingdorian** [comment]
The presets branch must be merged first.

**Comment by Kingdorian** [comment]
It should be fine now, however we still have to merge Naomi's presets branch into develop.

**Comment by jochem725** [comment]
Great, I'll agree with merging.

**Pull request merged**